### PR TITLE
Offcanvas:  activate focustrap when backdrop is enabled

### LINKED ISSUE #36596: Docs: problems with the two off-canvas components for docs navigation and general navigation on small screen
### Prerequisites

- [X] I have [searched](https://github.com/twbs/bootstrap/issues?utf8=%E2%9C%93&q=is%3Aissue) for duplicate or closed issues
- [X] I have [validated](https://html5.validator.nu/) any HTML to avoid common problems
- [X] I have read the [contributing guidelines](https://github.com/twbs/bootstrap/blob/main/.github/CONTRIBUTING.md)

### Describe the issue

Split from https://github.com/twbs/bootstrap/issues/36586 as I might have misunderstood what that issue was about, and proceeded to look at https://getbootstrap.com instead...and came across issues there:

it seems that for the navigation, something's not quite right in how the off-canvas was implemented: while the off-canvas for the "Toggle docs navigation" correctly traps focus, the off-canvas for the general navigation doesn't...users can still tab out of this and into the underlying page.

screenshot of open navigation off-canvas, but focus is visibly in the underlying page behind the semi-transparent overlay

@mdo worth checking how this was added/implemented in this case. Also yes, while the "Toggle navigation" button does have an `aria-expanded`, it's currently always set to `false`, regardless. I suspect this was a left-over from the previous design that used dropdowns, as the off-canvas doesn't set `aria-expanded` since the idea is that it's a modal, and the underlying page should not actually be accessible (because of the focus trapping and the `aria-modal="true"`).

Also noticed that focus is not moved to either of the off-canvases (the docs nav, and the general nav), but remains on the toggle buttons. Not looked at the markup in depth, but suspect a `tabindex="-1"` is missing on them (compare to the working off-canvas examples https://getbootstrap.com/docs/5.2/components/offcanvas/).

### Reduced test cases

N/A

### What operating system(s) are you seeing the problem on?

Windows

### What browser(s) are you seeing the problem on?

Chrome

### What version of Bootstrap are you using?

5.2

### PR DESCRIPTION
**Pull request by GeoSot**
closes #36596

Preview https://deploy-preview-36717--twbs-bootstrap.netlify.app/

**/cc @patrickhlauke**

### CODE REVIEW

**Comment by julien-deramond** [comment]
Added https://github.com/twbs/bootstrap/pull/36717/commits/658045e9aa70ecd9ce98c838e2c69b3acc171f7c to tackle the 3rd point from the issue:

> Also noticed that focus is not moved to either of the off-canvases (the docs nav, and the general nav), but remains on the toggle buttons. Not looked at the markup in depth, but suspect a tabindex="-1" is missing on them (compare to the working off-canvas examples https://getbootstrap.com/docs/5.2/components/offcanvas/).

To test it in the docs, simply focus the main header (keyboard), open both menus with 'Enter' and then press to 'Escape' to close them. 'Escape' didn't do anything before since offcanvases didn't get the focus. Now it's working.

**Comment by julien-deramond** [comment]
If I have well understood the 2nd point in the issue, it is safe to remove `aria-expanded="false"` from the `.navbar-toggler`s in the docs since offcanvases won't change the status and make an `aria-expanded="true"`.
Done via https://github.com/twbs/bootstrap/pull/36717/commits/d8f192dc9451957efb48057f43e98085c57270df.

**Review by julien-deramond — APPROVED** [review]
JS part LGTM! I've added 2 commits to handle 2 other points mentioned in the issue. This PR will need also a review from @patrickhlauke.

**Review by patrickhlauke — CHANGES_REQUESTED** [review]

**Comment by patrickhlauke** [comment]
Looking good, and correctly addresses the issue https://github.com/twbs/bootstrap/issues/36596

At a higher level though, I'm wondering if we should just always enable focus trapping? Or, if we *do* want to allow non-modal offcanvases, we should document this clearly in the docs (currently it seems the docs are silent about focus behaviour at all) and also make sure that `aria-modal="true"` is only added when it's intended to be a modal offcanvas

File: js/tests/unit/offcanvas.spec.js [hunk 1]
```diff
@@ -306,6 +307,27 @@ describe('Offcanvas', () => {
         offCanvas.show()
       })
     })
+
+    it('should trap focus if scroll is allowed & backdrop is enabled', () => {
```
Comment (GeoSot) [inline js/tests/unit/offcanvas.spec.js:311]:
```suggestion
    it('should trap focus if scroll is allowed OR backdrop is enabled', () => {
```

**Comment by GeoSot** [comment]
>  At a higher level though, I'm wondering if we should just always enable focus trapping? Or, if we do want to allow non-modal offcanvases, we should document this clearly in the docs (currently it seems the docs are silent about focus behaviour at all) and also make sure that aria-modal="true" is only added when it's intended to be a modal offcanvas


Either way, an offcanvas that doesn't scroll or has a backdrop, behaves like modal, so I think is right for know.
BUT yes, we may, can discuss it for next releases

**Review by patrickhlauke — APPROVED** [review]

**Pull request merged**